store/sql_post_store.go

```diff
@@ -471,6 +473,8 @@ func (s SqlPostStore) Search(teamId string, userId string, terms string, isHasht
 				    ORDER BY CreateAt DESC
 				LIMIT 100`, searchType)
 
+			terms = strings.TrimRight(terms, specialSearchCharCutset)
```

Comment:
@jwilander Can we add the above cases as unit tests.  I'm 99% sure they will fail in postgres too.  
"(hello)corey"
"(hello)"